src/Raven.Server/Documents/Indexes/MapReduce/OutputToCollection/ReduceOutputDocumentActions.cs

```diff
@@ -51,6 +65,34 @@ public void Initialize(RavenTransaction tx)
             }
         }
 
+        public void AddPatternGeneratedIdForReduceOutput(Transaction tx, string reduceResultId, string referenceDocumentId)
+        {
+            var tree = tx.CreateTree(ReduceOutputsIdsToPatternReferenceIdsTreeName);
+
+            tree.Add(reduceResultId, referenceDocumentId);
+        }
+
+        public string GetPatternGeneratedIdForReduceOutput(Transaction tx, string reduceResultId)
+        {
+            var tree = tx.CreateTree(ReduceOutputsIdsToPatternReferenceIdsTreeName);
+
+            var result = tree.Read(reduceResultId);
+
```

Comment:
can this be null?

Reply:
Should never happen. I'll add explicit exception for that